README.md

```diff
@@ -0,0 +1,246 @@
+- [x] Provide custom validation and serialization rules on a per-field basis for each model
+- [x] Support any API on a per-model basis with configurable serializers and API adapters, with default support for a standard REST API
+- [x] Sensible default configuration with plenty of escape hatches to suit your needs
+- [x] Feel confident with an open source library that is thoroughly unit tested
+- [x] Rendering library/framework agnostic:
+  - [x] **Official support for React** - [redux-data-service-react](https://github.com/Rediker-Software/redux-data-service-react)
+  - [ ] Vue: TBD
+  - [ ] Angular: TBD
+
+## Create models with `TypeScript` classes:
+
+```typescript
+// student/model.ts
+import { attr, required, Model, EmailField, NumberField, StringField } from "redux-data-service";
+
+export class Student extends Model {
+  public readonly string serviceName = "student";
+  
+  @required()
+  @attr(StringField)
+  public string firstName;
+  
+  @attr(NumberField)
+  public string age;
+  
+  @attr(EmailField)
+  public string email;
+}
+```
+
+* Validation and serialization rules are provided using property decorators.
+* Models can also support `belongsTo` and `hasMany` relationships to other models which are lazy-loaded when they are requested.
+
+## Interact with models through a familiar syntax
+
+```typescript
+import { Student } from "./student";
+
+const student = new Student({ id: "123" });
+
+student.firstName = "Jessica";
+student.age = 9;
+student.email = "jess@example.com";
+```
+
+* In preserving the [awesome power of an immutable architecture](https://youtu.be/rtcn9I9sB5M?t=22m5s), the above will not mutate the model.
+* Instead, it uses magic setters to dispatch model-specific actions, where a new instance of the model is created each time:
+
+```typescript
+{ type: "student/SET_FIELD", payload: { id: "123", fieldName: "firstName", value: "Jessica" } }
+{ type: "student/SET_FIELD", payload: { id: "123", fieldName: "age", value: 9 } }
+{ type: "student/SET_FIELD", payload: { id: "123", fieldName: "email", value: "jess@example.com" } }
+```
+
+Commit pending changes to the model's corresponding API end-point:
+
+```typescript
+student.save();
+```
+
+Calling the `save` method validates the model and dispatches an action to an underlying [redux-observable](https://redux-observable.js.org/) epic:
+
+```typescript
+{ type: "student/UPDATE_RECORD", payload: { id: "123" } }
+```
+
+The epic will then:
+
+* serialize the model into a JSON string (using the default serializer)
+* send a `POST` or a `PUT` to the API to create or update the model (using the default API adapter):
+
+```typescript
+PUT api/students/123
+{"id": "123", "firstName": "Jessica", "age": 9, "email": "jess@example.com"}
+```
+
+> Although the above example uses the default serializer and API adapter for connecting to a REST API, any `ISerializer` or `IAdapter` may be used to connect to *any* API.
```

Comment:
I would link to our `ISerializer.ts` and `IAdapter.ts` so the reader can have an example to go by.